woocommerce/woocommerce
# [CYS] Fix the selected pattern in footer, header and homepage

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





The problem was that the comparison to select a pattern was to compare their block attributes. We had 2 patterns with the same attributes so when selecting the last one it always matched the first.
In this PR I'm adding a specific class to each of the patterns to make their attributes different. Also, I've added an extra comparison to check if the number of inner blocks is the same.

Closes https://github.com/woocommerce/woocommerce/issues/45215



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure the `WooCommerce Beta Tester` plugin is installed and activated (available on this monorepo).
2. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
3. Click on `Tools` and run the `Reset Customize Your Store` and the `Delete all products` commands.
4. Go to `WooCommerce > Home`, and click on `Start customizing`.
5. In the assembler, click on the `Choose your footer`, select a pattern, and go back by clicking on the arrow.
6. Repeat 5 for the other 2 footer patterns.
7. Repeat 5-6 steps also for the `Choose your header`.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
CYS - Fix the selected pattern in footer in the assembler.
#### Comment 

</details>
